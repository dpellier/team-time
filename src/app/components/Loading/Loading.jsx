
import React from 'react';

const Loading = () =>
    <div className='loading'>
        <svg viewBox='0 0 40 40'>
            <g>
                <defs>
                    <linearGradient id='spinner' gradientUnits='userSpaceOnUse' x1='34' y1='29' x2='4' y2='29'>
                        <stop offset='.1' stopColor='#35bdfd' stopOpacity='0' />
                        <stop offset='1' stopColor='#35bdfd' />
                    </linearGradient>
                </defs>

                <g strokeWidth='3' fill='none'>
                    <path stroke='#35bdfd'
                          d='M3,20 A17,17 0 0,1 26,4'>
                    </path>

                    <path stroke='url(#spinner)'
                          d='M37,20 A1,1 0 0,1 3,20'>
                    </path>

                    <animateTransform values='0,20,20;360,20,20'
                                      attributeName='transform'
                                      type='rotate'
                                      repeatCount='indefinite'
                                      dur='1s'>
                    </animateTransform>
                </g>
            </g>
        </svg>
    </div>
;

export { Loading };
