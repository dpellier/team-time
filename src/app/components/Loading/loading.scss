
.loading {
    position: absolute;
    top: 50%;
    right: 0;
    left: 0;
    transform: translateY(-50%);
    margin: auto;
    width: 80px;
    height: 80px;
}
