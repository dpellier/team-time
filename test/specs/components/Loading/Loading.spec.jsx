
import React from 'react';
import renderer from 'react-test-renderer';
import { Loading } from '../../../../src/app/components/Loading/Loading';

describe('<Loading />', () => {
    it('is correctly rendered', () => {
        const component = renderer.create(
            <Loading />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
