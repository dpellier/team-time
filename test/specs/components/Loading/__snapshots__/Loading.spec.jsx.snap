// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> is correctly rendered 1`] = `
<div
  className="loading"
>
  <svg
    viewBox="0 0 40 40"
  >
    <g>
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="spinner"
          x1="34"
          x2="4"
          y1="29"
          y2="29"
        >
          <stop
            offset=".1"
            stopColor="#35bdfd"
            stopOpacity="0"
          />
          <stop
            offset="1"
            stopColor="#35bdfd"
          />
        </linearGradient>
      </defs>
      <g
        fill="none"
        strokeWidth="3"
      >
        <path
          d="M3,20 A17,17 0 0,1 26,4"
          stroke="#35bdfd"
        />
        <path
          d="M37,20 A1,1 0 0,1 3,20"
          stroke="url(#spinner)"
        />
        <animateTransform
          attributeName="transform"
          dur="1s"
          repeatCount="indefinite"
          type="rotate"
          values="0,20,20;360,20,20"
        />
      </g>
    </g>
  </svg>
</div>
`;
